### PR TITLE
ODK in GitHub action

### PR DESCRIPTION
**Pull request by joeflack4**
## Overview
Setting up ODK for GitHub actions so that we can run in a good environment with all dependencies, e.g. `robot`.

## Updates
ODK-based GitHub Action
    - Update: GH action workflow: (i) now uses ODK container, (ii) Had to change 'echo -en' to plain 'echo' due to some new incompatibility with new ODK environment, (iii) removed some comments; streamlined.
    - Update: Minor updates in a few files; code comments, codestyle, verbosity

## Depends on
- https://github.com/INCATools/ontology-development-kit/issues/888

## Related issues
- #92 
- #97

### CODE REVIEW

File: .github/workflows/buid_and_release.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ on:
 jobs:
   build_and_release:
     runs-on: ubuntu-latest
+    container: obolibrary/odkfull:v1.3.0
```
Comment (joeflack4) [inline .github/workflows/buid_and_release.yml:16]: The main line I changed.

File: .github/workflows/buid_and_release.yml [hunk 2]
```diff
@@ -34,20 +35,23 @@ jobs:
   #      utcOffset: "+08:00"
         format: YYYY-MM-DD
     - name: build
-      run: make automated-release-artefacts -B
-    - name: Release
-      run: echo Uploading files as new release.
-# This one wouldn't work unless it was a tag and on-push.
-#      uses: softprops/action-gh-release@v1
+#      run: make automated-release-artefacts -B
+      run: robot --help
```
Comment (joeflack4) [inline .github/workflows/buid_and_release.yml:39]: Just temporarily doing a face check here.

File: .github/workflows/buid_and_release.yml [hunk 3]
```diff
@@ -13,15 +13,18 @@ on:
 jobs:
   build_and_release:
     runs-on: ubuntu-latest
+#    container: obolibrary/odkfull:v1.4.1
+#    container: obolibrary/odkfull:dev
+    container: obolibrary/odkfull:latest
```
Comment (matentzn) [inline .github/workflows/buid_and_release.yml:18]: Always use a version to minimise reasons of failure.

File: .github/workflows/buid_and_release.yml [hunk 4]
```diff
@@ -13,15 +13,18 @@ on:
...
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python 3.10
       uses: actions/setup-python@v2
```
Comment (matentzn) [inline .github/workflows/buid_and_release.yml:22]: No need to set up python, its already in ODK

File: .github/workflows/buid_and_release.yml (hunk 3 above)
Comment (joeflack4) [reply .github/workflows/buid_and_release.yml:18]: @matentzn Yeah I agree with that. I was thinking I should probably change that too. Glad you agree!

File: .github/workflows/buid_and_release.yml (hunk 4 above)
Comment (joeflack4) [reply .github/workflows/buid_and_release.yml]: Yeah I overlooked that and found that out the hard way. It was pointed out to me as well by: https://github.com/INCATools/ontology-development-kit/issues/888

File: .github/workflows/buid_and_release.yml [hunk 5]
```diff
@@ -6,48 +6,41 @@ name: Build and release
...
-      with:
-        python-version: '3.10'
     - name: Install
       run: |
-        python -m pip install --upgrade pip
         make install
     - name: Create .env file
       run: |
-        echo -en "API_KEY=${{ secrets.API_KEY }}" > .env
+        echo "API_KEY=${{ secrets.API_KEY }}" > .env
```
Comment (joeflack4) [inline .github/workflows/buid_and_release.yml:22]: A bugfix. For some reason `echo -en` wasn't working for me after switching over to the ODK container.

File: .github/workflows/buid_and_release.yml [hunk 6]
```diff
@@ -6,48 +6,41 @@ name: Build and release
...
-#    - cron: "0 0 1 * *"  # monthly
   workflow_dispatch:
 
 jobs:
   build_and_release:
     runs-on: ubuntu-latest
+    container: obolibrary/odkfull:v1.4.1
     steps:
     - uses: actions/checkout@v2
-    - name: Set up Python 3.10
```
Comment (joeflack4) [inline .github/workflows/buid_and_release.yml:18]: Removed this step and it fixed the ODK issue I was having, as suggested.

File: .github/workflows/buid_and_release.yml [hunk 7]
```diff
@@ -6,48 +6,41 @@ name: Build and release
...
   schedule:
     - cron: "0 0 * * 0"  # weekly on Sunday at midnight
-#    - cron: '0 0 * * *'  # every day at midnight
-#    - cron: "0 0 1 * *"  # monthly
   workflow_dispatch:
 
 jobs:
   build_and_release:
     runs-on: ubuntu-latest
+    container: obolibrary/odkfull:v1.4.1
```
Comment (joeflack4) [inline .github/workflows/buid_and_release.yml:14]: Locked in to `1.4.1`

**Review by joeflack4 — COMMENTED** [review]
@matentzn This is ready for your review now. I was able to successfully run it just now.

**Pull request merged**